extensions/mouse/internal.m

```diff
@@ -114,7 +114,7 @@ -(io_service_t)createIOHIDSystem {
 }
 
 -(NSDictionary *)getIOHIDParamtersFromService:(io_service_t)service {
-    return (__bridge NSDictionary *)IORegistryEntryCreateCFProperty(service, CFSTR(kIOHIDParametersKey), kCFAllocatorDefault, kNilOptions);
+    return CFBridgingRelease(IORegistryEntryCreateCFProperty(service, CFSTR(kIOHIDParametersKey), kCFAllocatorDefault, kNilOptions));
```

Comment:
The constructs aren't quite identical -- `(__bridge NSDictionary *)` doesn't change ownership; `CFBridgingRelease` does.

I haven't looked at the rest of the code, so maybe we were wrong to begin with, but I did want to point out the change in semantics just in case you also need to add something elsewhere to clean it up.

Reply:
From looking at the analysis, I think that change of ownership is needed - `IORegistryEntryCreateCFProperty()` is a Create under the CF ownership rules, so we are responsible for freeing its return, but we just bridge it to `NSDictionary` and nothing else, but by changing it to a `CFBridgingRelease()`, it's ARC's responsibility and thus returning it as an `NSDictionary` is fine.....

..... right? :)

Reply:
Sounds right to me.